### PR TITLE
[oneDNN] Enable Conv2D+FBNV2/V3+activation fusion for bf16

### PR DESCRIPTION
**Pull request by gaurides**
This PR enables the fusion disabled by [PR-53867](https://github.com/tensorflow/tensorflow/pull/53867 )
This PR adds attribute U to FusedConv2D so as to account for FusedBatchNormV2/V3 inputs that are of type U. With that, the graph generated after remapper is valid and all grappler optimizations stay after MetaOptimizer is executed.

### CODE REVIEW

**Comment by gaurides** [comment]
[@elfringham](https://github.com/elfringham) : do you know why ARM CI failed for this PR?

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Review by ezhulenev — APPROVED** [review]

**Comment by gaurides** [comment]
@ezhulenev - can you please re-approve this PR? 
I had to merge master to this PR to address some unit-test failure fixed in master. So, it needs re-approval. Thanks!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Review by ezhulenev — APPROVED** [review]

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc [hunk 1]
```diff
@@ -1521,6 +1521,15 @@ class MklFusedConvOp
           context, num_args == 4,
           errors::InvalidArgument(
               "Fused Conv2D with batchnorm must have 4 extra argument"));
+      std::vector<DataType> TArgs;
+      OP_REQUIRES_OK(context, context->GetAttr("TArgs", &TArgs));
+      OP_REQUIRES(
+          context, TArgs.size() == 4,
+          errors::InvalidArgument("FusedConv2D TArgs size expected to be 4"));
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_conv_ops.cc:1528]: Please replace `errors::InvalidArgument` with `absl::InvalidArgumentError`. Same for other kinds of `errors::*`. You can search for example usage in the repo, e.g.,
https://github.com/tensorflow/tensorflow/blob/b7f213c61d79dfaa359b0e0de22e773e2fcb538c/tensorflow/cc/framework/ops.h#L161-L163

Please do this for all files in this PR and in future PRs.

**Review by penpornk — CHANGES_REQUESTED** [review]
We got internal code check failures. We don't use `tsl::errors::*` anymore, they need to be replaced with their `absl` counterparts. Would you mind fixing this in all files? Thank you!

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc (hunk 1 above)
Comment (gaurides) [reply tensorflow/core/kernels/mkl/mkl_conv_ops.cc]: @penpornk  - I changed all occurances of errors::* in mkl_conv_ops.cc. Also, needed to use absl::StrCat() and adjust formatting due to which you might see more LOC. No errors::* used in any other files in this PR.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the changes! I have one formatting nit (or this wouldn't pass the code check).

File: tensorflow/core/kernels/mkl/mkl_conv_ops.cc [hunk 2]
```diff
@@ -862,8 +862,8 @@ class MklConvOp : public OpKernel {
         // Inputs to FusedBatchNorm have same 1D shape
         fuse_bn_shape = MklGetInput(context, kInputIndex_BN_Mean).shape();
         OP_REQUIRES(context, fuse_bn_shape.dims() == 1,
-                    errors::InvalidArgument("FusedBatchNorm must be 1D, not: ",
-                                            fuse_bn_shape.DebugString()));
+                    absl::InvalidArgumentError(absl::StrCat("FusedBatchNorm must be 1D, not: ",
```
[the remark this answers is not in the document]
Comment (gaurides) [reply tensorflow/core/kernels/mkl/mkl_conv_ops.cc]: Oh, sorry about that. I thought I did. Now it should be good.

**Comment by gaurides** [comment]
I am working to resolve conflicts

**Comment by gaurides** [comment]
@penpornk - it shows the Py+CPP Internal CI build failed. I looked at the public invocation log, the failures don't seem related to my changes. Are these known issues? Can you please point me if there's any failure that needs to be fixed?

**Review by penpornk — APPROVED** [review]
Thank you for the fixes. I agree the PY+CPP CPU and GPU failures aren't related to this PR. They seem more like infra failures (all the failed tests timed out).

File: tensorflow/core/grappler/optimizers/remapper.cc [hunk 3]
```diff
@@ -945,14 +945,9 @@ bool FindConv2DWithBatchNorm(const RemapperContext& ctx, int node_index,
   if (!IsFusedBatchNorm(*node_def)) return false;
 
   // FusedBatchNormV2 and V3 have an extra type parameter.
-  // Conv2D + FusedBatchNormV2/V3 fusion is currently not supported for bf16.
-  // TODO(intel-tf): enable the fusion for bf16
   bool dtypeU_is_float = HasDataType(node_def, DT_FLOAT, "U");
   bool dtypeT_is_bf16 = HasDataType(node_def, DT_BFLOAT16, "T");
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/remapper.cc:949]: Please remove this unused variable. Thank you!

**Review by penpornk — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/optimizers/remapper.cc (hunk 3 above)
Comment (gaurides) [reply tensorflow/core/grappler/optimizers/remapper.cc:949]: thanks. done.

**Review by penpornk — APPROVED** [review]
Thank you!

File: tensorflow/core/kernels/fused_batch_norm_op.cc [hunk 4]
```diff
@@ -1963,6 +1963,7 @@ REGISTER_KERNEL_BUILDER(Name("FusedBatchNormGradV3")
                             .TypeConstraint<float>("U"),
                         FusedBatchNormGradOpV3<CPUDevice, Eigen::half, float>);
 
+
```
Comment (penpornk) [inline tensorflow/core/kernels/fused_batch_norm_op.cc:1966]: Is this newline added by clang-format? If it didn't break any formatting checks, I don't think we need a newline here.

Comment (gaurides) [reply tensorflow/core/kernels/fused_batch_norm_op.cc:1966]: sorry, i added by mistake on this PR instead of another ... reverting now.

Comment (gaurides) [reply tensorflow/core/kernels/fused_batch_norm_op.cc]: i have fixed it now.

**Comment by gbaned** [comment]
Hi @gaurides Can you please rebase and resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @penpornk Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @gaurides Can you please rebase your branch and resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @gaurides Can you please rebase your branch and resolve the conflicts? Thank you!

**Comment by gaurides** [comment]
> Hi @gaurides Can you please rebase your branch and resolve the conflicts? Thank you!

thanks @gbaned. I just rebase the branch and resolved the conflict.

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @gaurides Can you please confirm if this PR is still valid? Thank you!

**Comment by gbaned** [comment]
Hi @gaurides Can you please confirm if this PR is still valid? Thank you!

**Comment by gaurides** [comment]
As of now, it looks like it is not required. So I will close this PR.

**Pull request closed without merging**